### PR TITLE
Ch2527/search call

### PR DESCRIPTION
**Pull request by qbasso**

### CODE REVIEW

**Comment by qbasso** [comment]
@rubikzube updated

File: library/src/main/java/io/constructor/data/model/search/Option.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package io.constructor.data.model.search
+
+data class Option(val count: Int, val value: String?)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/Option.kt]: Can we rename to facet option?

File: library/src/main/java/io/constructor/data/model/search/Result.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+import io.constructor.data.model.Group
+import io.constructor.data.model.ProductData
+
+data class Result(val description: String?, val id: String,
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/Result.kt]: We should discuss typing tomorrow.  Some of these properties are universal and some are not :(

File: library/src/main/java/io/constructor/data/model/search/Path.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+import io.constructor.data.model.ProductData
+
+data class Path(@Json(name = "display_name") val displayName: String, val id: Long)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/Path.kt]: Not sure what this is

File: library/src/main/java/io/constructor/data/model/search/SearchResponse.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+
+data class SearchResponse(@Json(name = "response") val searchData: SearchData, @Json(name = "result_id") val resultId: String, var rawData: String?)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/SearchResponse.kt]: Looks good.

File: library/src/main/java/io/constructor/data/model/AutocompleteResult.kt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package io.constructor.data.model
+
+
+data class AutocompleteResult(val sections: Sections)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/AutocompleteResult.kt]: Can we add resultID to this class?

Comment (rubikzube) [reply library/src/main/java/io/constructor/data/model/AutocompleteResult.kt]: What are your thoughts on renaming to `AutocompleteResponse`?

File: library/src/main/java/io/constructor/data/model/Group.kt [hunk 6]
```diff
@@ -1,7 +1,8 @@
 package io.constructor.data.model
 
 import com.squareup.moshi.Json
+import io.constructor.data.model.search.Path
 import java.io.Serializable
 
 
-data class Group(@Json(name="display_name") var displayName: String?, @Json(name="group_id") var groupId: String, @Json(name="path") var path: String? = null) : Serializable
\ No newline at end of file
+data class Group(@Json(name="display_name") var displayName: String?, @Json(name="group_id") var groupId: String, @Json(name="path") var path: String? = null, @Json(name="path_list") var pathList: List<Path>? = null) : Serializable
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/Group.kt]: Let's rename to `ResultGroup`... to differentiate from the rolled up groups with counts at the top level of a search response?

File: library/src/main/java/io/constructor/data/model/search/Group.kt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+import io.constructor.data.model.ProductData
+
+data class Group(val value: String, val data: ProductData, @Json(name = "matched_terms") val matchedTerms: List<String>?)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/Group.kt]: Not sure what this is used for.

File: library/src/main/java/io/constructor/data/model/search/GroupData.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+
+data class GroupData(@Json(name = "children") val data: List<String>, val count: Int, @Json(name = "display_name") val displayName: String,
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/GroupData.kt]: Let's rename to `FilterGroup`?

File: library/src/main/java/io/constructor/data/model/search/Facet.kt [hunk 9]
```diff
@@ -0,0 +1,3 @@
+package io.constructor.data.model.search
+
+data class Facet(val name: String, val options: List<Option>?)
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/Facet.kt]: Looks good

File: library/src/main/java/io/constructor/data/model/search/Option.kt (hunk 1 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/Option.kt]: its under `search` package so I thought its enough, but sure

File: library/src/main/java/io/constructor/data/model/search/Result.kt (hunk 2 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/Result.kt]: not sure if I get this one

File: library/src/main/java/io/constructor/data/model/search/ResultData.kt [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+import io.constructor.data.model.ProductData
+
+data class ResultData(@Json(name = "data") val result: Result, @Json(name = "matched_terms") val matchedTerms: List<String>?, val value: String)
```
[the remark this answers is not in the document]
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/ResultData.kt]: ok

File: library/src/main/java/io/constructor/data/model/search/Path.kt (hunk 3 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/Path.kt]: its part of the model, guess you guys should know what it is :)

File: library/src/main/java/io/constructor/data/model/search/Group.kt (hunk 7 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/Group.kt]: must have been development leftover, good call

File: library/src/main/java/io/constructor/data/model/AutocompleteResult.kt (hunk 5 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/AutocompleteResult.kt]: yes, sounds better

File: library/src/main/java/io/constructor/data/model/search/Path.kt (hunk 3 above)
Comment (rubikzube) [reply library/src/main/java/io/constructor/data/model/search/Path.kt]: I don't think it's used?

File: library/src/main/java/io/constructor/data/model/search/ResultData.kt [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package io.constructor.data.model.search
+
+import com.squareup.moshi.Json
+import io.constructor.data.model.ResultGroup
+
+data class ResultData(val description: String?, val id: String,
+                      @Json(name = "image_url") val imageUrl: String?,
+                      val price: Double?,
+                      val quantity: String?,
+                      val url: String?,
```
Comment (rubikzube) [inline library/src/main/java/io/constructor/data/model/search/ResultData.kt]: Re: slack, `price` and `quantity` could exist or not exist per customer.

File: library/src/main/java/io/constructor/data/model/search/Path.kt (hunk 3 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/Path.kt]: I don't use it anywhere in SDK, added model to fully parse response

File: library/src/main/java/io/constructor/data/model/search/ResultData.kt (hunk 11 above)
Comment (qbasso) [reply library/src/main/java/io/constructor/data/model/search/ResultData.kt]: price and qunatity are already nullable, so their their  presence (or not) won't crash anything. Should I add this `slack` property?

Comment (rubikzube) [reply library/src/main/java/io/constructor/data/model/search/ResultData.kt]: Nah, `slack` referred to actual Slack.  Like the app Slack.  Is there any way to make ResultData have a mix of typed and untyped properties?

File: library/src/main/java/io/constructor/data/model/search/Path.kt (hunk 3 above)
Comment (rubikzube) [reply library/src/main/java/io/constructor/data/model/search/Path.kt]: Where is it used to parse the response?  Couldn't find a reference to it.

**Pull request closed without merging**